chalk-parse/src/ast.rs

```diff
@@ -175,30 +302,60 @@ impl PolarizedTraitRef {
     }
 }
 
+/// A valid Rust identifier, along with a `Span` specifying line numbers.
 #[derive(Copy, Clone, Debug)]
 pub struct Identifier {
     pub str: InternedString,
     pub span: Span,
 }
 
+/// A where clause; either a normal Rust trait bound, or a Chalk-specific clause.
+// TODO: Add a "See" that points to a list of Chalk's where clauses, and a
+// description of each.
```

Comment:
Maybe link here?

https://rust-lang-nursery.github.io/rustc-guide/traits-lowering-rules.html#lowering-where-clauses

Reply:
Woah! I've never seen that guide before!